### ISSUE TITLE
Remember me not working after upgrade to 1.2.20 from 1.2.19

### ISSUE THREAD

**Issue by bcoory** [Cacti/cacti#4705]
# Describe the bug
After upgrading to 1.2.20 from 1.2.19 I cannot get the remember me function to work after closing the browser. Tried on multiple browsers same issue.

Confirmed Support Authentication Cookies is enabled.

Cleared all cookies and cache from browser.

Cookie info in broswer says it Expires When the browsing session ends.

Rolled back to 1.2.19 to test and works as expected.

Issue occurs with both local and LDAP login.

Single Cacti instance - Tried modifying cacti_session_name with no luck.

cacti_cookie_domain is commented out. I also tried enabling this with our Cacti FQDN with no luck.

## To Reproduce

Upgrade and login ticking "Keep me signed in". Save password when prompted by browser. Then close browser and go to cacti site.

## Expected behavior
Bypass login page once "Keep me signed in" is ticked.

## Desktop (please complete the following information)

- OS: Windows 10 Pro 21H2

- Browser: Latest Chrome, Edge and Firefox

- Version [e.g. 22] Chrome- 100.0.4896.75 Edge- 00.0.1185.39 Firefox- 99.01

**Comment by TheWitness** [comment]
This is duplicate.  Try updating to the latest 1.2.x branch and see if the problem goes away.

**Comment by TheWitness** [comment]
Another note is that we did a change to the Remember Me support.  After you grab the latest goto your User Profile and press the logout everywhere button.  If you want to do that for everyone, you simply have to `TRUNCATE TABLE user_auth_cache;`.  There was a bug prior to 1.2.20 where the tokens were not realm away, say LDAP or Local.  So, that was added to the `user_auth_cache` table.

**Comment by bcoory** [comment]
I have just tried updating to the latest 1.2.x branch and using the "Log out everywhere" button. Still having the same issue.

**Comment by TheWitness** [comment]
What auth method are you using?

**Comment by bcoory** [comment]
LDAP. But i get the same issue using a local account.

**Comment by TheWitness** [comment]
Then, you must not be on the latest or there is some other reason.  Did you truncate the table?  That might fix it.

**Comment by bcoory** [comment]
Definitely on the latest. I checked the table before truncating and it was empty. After that I tried logging in using both a local and AD account and ticking remember but nothing is written to the table.

**Comment by TheWitness** [comment]
Well that's odd.  I can not reproduce.  Can you check your browsers debug panel to see if the cookie is being rejected?  Are you using https?

**Comment by bcoory** [comment]
Cookie is there - Using http only

**Comment by TheWitness** [comment]
Okay, that might be it.  Let me see if I can reproduce that way.

**Comment by bcoory** [comment]
I just tried https, same problem. Nothing in the table yet either. Deleted cookie and it gets recreated ok.

**Comment by TheWitness** [comment]
It's http that's breaking it.  On https it's working fine.  Let me look into that.  I've been using https for so long as it's where just about everyone is these days.

**Comment by TheWitness** [comment]
Okay, pull a fresh copy and then remove your cookies before you try again (just in case).

**Comment by bcoory** [comment]
Odd. Still having the same issue with both http and https after the changes.

**Comment by TheWitness** [comment]
Using Firefox, you should be able to open the debug window, goto Storage, drill into cookie storage and your view should look like the following.  Not the columns that I've highlighted.  For this to work 'correctly' you need to be at PHP7.3++

![image](https://user-images.githubusercontent.com/1439914/163398834-5c5ba12d-e45a-4508-b00f-6f272cad24b5.png)

**Comment by TheWitness** [comment]
If they are not as specified, and you are at PHP7.3+, then delete them one at a time and start over.

**Comment by bcoory** [comment]
Running PHP7.4. I am running FPM with apache. Will try without FPM also

**Comment by bcoory** [comment]
It seems the cacti_remembers cookie is not being created. Happening for me in all browsers. HTTP and HTTPS. 

![image](https://user-images.githubusercontent.com/21350148/163670050-1ff32b65-969a-46d7-8b18-3337b7fb94b7.png)

**Comment by TheWitness** [comment]
It may seem to be a stupid question, but it is enabled right?  I'm totally unable to reproduce.

![image](https://user-images.githubusercontent.com/1439914/163813164-8f7fdf59-ea2f-4395-80be-6e50122e2ff3.png)

**Comment by batman978** [comment]
Hello, I can confirm this is happening for me too - Firefox and Edge.  I don't see 'cacti_rememebers' cookie being created.
I upgraded 1.2.19 to 1.2.20 using PHP 7.4.3 on Ubuntu 20.04.  My desktop is Windows 10 Enterprise 21H2

**Comment by batman978** [comment]
Ahh ... "logout everywhere" worked for me.  Sorry I missed that higher in the thread.
I only did it for my account, I didn't do it against the user_auth_cache table.

**Comment by bcoory** [comment]
> It may seem to be a stupid question, but it is enabled right? I'm totally unable to reproduce.
> 
> ![image](https://user-images.githubusercontent.com/1439914/163813164-8f7fdf59-ea2f-4395-80be-6e50122e2ff3.png)

Yes, definitely turned on. 

I cannot see any errors in the the console showing the cookie being rejected.

**Comment by TheWitness** [comment]
@batman978, did you update to the 1.2.x branch?  Are you using http or https?

**Comment by TheWitness** [comment]
@bcoory, what you need to do it goto the login page, bring up the debug panel, and then the network tab, and watch the request and response headers.  See below.

![KeepMeSignedIn](https://user-images.githubusercontent.com/1439914/164110938-2e191a86-58ef-4c1d-b8f1-a95662e76166.gif)

**Comment by bcoory** [comment]
There seems to be no response cookie.

https://user-images.githubusercontent.com/21350148/164124988-fc82133c-2dce-4690-af48-b7baa97df4f7.mov

**Comment by TheWitness** [comment]
That really helped.  Login, goto `Console > Configuration > Settings > Authentication` and make your Guest Account different something different than `admin`.

**Comment by TheWitness** [comment]
I guess we should be blocking the `admin` account from being a `guest` account.

**Comment by TheWitness** [comment]
Logged this bug as a result of the finding.  Thanks!

https://github.com/Cacti/cacti/issues/4731

**Comment by bcoory** [comment]
I currently have no guest account. Only one local "admin" account and 2 LDAP users.

![image](https://user-images.githubusercontent.com/21350148/164127409-e255ce33-1186-4db8-86f0-06f71e63d940.png)

**Comment by TheWitness** [comment]
Starts at line 201.

**Comment by bcoory** [comment]
Results below. Tried both local and LDAP.

2022/04/21 08:39:42 - AUTH Username error, not setting token
2022/04/21 08:39:42 - AUTH LOGIN: User 'admin' authenticated
2022/04/21 08:39:05 - AUTH Username error, not setting token
2022/04/21 08:39:05 - AUTH LOGIN: User 'bcoory' authenticated
2022/04/21 08:39:05 - AUTH LOGIN: LDAP User 'bcoory' Authenticated
2022/04/21 08:39:05 - AUTH LDAP_SEARCH: Authentication Success,

**Comment by TheWitness** [comment]
Are you running any plugins?  I'm betting we have a name space collision.

**Comment by TheWitness** [comment]
Or it could be you simply need to go to the `Console > Configuration > Settings > Authentication` page and set a guest account, hit save, and then try to login again.

**Comment by TheWitness** [comment]
It's not a namespace issue though after looking at the error.

**Comment by bcoory** [comment]
I created a guest account and assigned it on Console > Configuration > Settings > Authentication. Now it is working.


I have never had a guest account in the past.

**Comment by TheWitness** [comment]
It's kind of crazy.  Likely what happened before, is that it was set that way, and you never realized it.  It's good you have it set now as your Cacti basically had some areas that if a user went to it were basically unsecured.  This is why we prevented the 'guest' account, moving forward from being the primary admin account.  I guess what we missed in the upgrade was 'fixing' it if we found it.  Glad it's working for you now.

**Comment by bcoory** [comment]
Many thanks! It looks like I can disable the guest account and remember me is still working.

**Comment by KnoAll** [comment]
I'm running into a similar issue. I've followed through some of the troubleshooting on this page.
Running v1.2.21 on PHP 7.3.x and 7.4.x. on two different systems both with this issue.
I'm running local realm on HTTP only. 
I've checked that Support Auth Cookies is enabled. 
Ran a login with the debug/network/cookies, and do not see any response cookies entries.
I have a guest account on both systems, but don't use it/ it is not enabled.
Any help would be appreciated, happy to provide additional info as needed.
